### DISCUSSION TITLE
Compiling fpm with GCC 5

### DISCUSSION THREAD

**Discussion by awvwgk** [fortran-lang/fpm#393] [General]
Anyone has experience building fpm with GCC 5 / MinGW on Windows? I know TOML Fortran is GCC 5 compatible, not sure about fpm itself though. Being able to compile with GCC 5 would enable us to ship Windows binaries on conda-forge.

**Comment by arjenmarkus** [comment]
GCC 5 dates from 2015 - about half an eternity in software terms. There is
a version of MinGW which has GCC 10.2.0 - I must admit that my
understanding of the distributions of MinGW is limited, but an up-to-date
installation should have at the very least have access to that version.

Op di 16 mrt. 2021 om 10:42 schreef Sebastian Ehlert <
***@***.***>:

**Reply by awvwgk** [reply]
Sadly, GCC 5 / MinGW is still the tool chain available on conda-forge for Windows, the alternative is flang 11, which is broken for sure with TOML Fortran. Maybe conda-forge is not the best way to distribute fpm for Windows, but I didn't get much feedback on Windows distributions when I asked on the discourse recently (https://fortran-lang.discourse.group/t/distributing-fortran-projects-for-proprietary-oses/760).

**Comment by arjenmarkus** [comment]
Hm, I saw the title but misunderstood it apparently - I did not associate
"proprietary" with "Windows" or "MinGW" or the like. I do not know much
about conda-forge, but there ought to be alternatives ... (Or should I say:
there should?)

Op di 16 mrt. 2021 om 11:10 schreef Sebastian Ehlert <
***@***.***>:

**Reply by awvwgk** [reply]
I *may* have grouped OSX and Windows as proprietary operating systems.

**Comment by LKedward** [comment]
It looks like GCC 5 isn't enough for *fpm*, I get the following:

```
././src/fpm_command_line.f90:191:12:

             cmd_settings=fpm_run_settings(&
            1
Error: Assignment to an allocatable polymorphic variable at (1) is not yet supported
```
```
$ gfortran --version
GNU Fortran (Ubuntu 5.5.0-12ubuntu1) 5.5.0 20171010
Copyright (C) 2015 Free Software Foundation, Inc.

GNU Fortran comes with NO WARRANTY, to the extent permitted by law.
You may redistribute copies of GNU Fortran
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING
```
